contracts/proxy/contract.go

```diff
@@ -42,8 +53,69 @@ func Update(nefFile, manifest []byte, data any) {
 
 // Verify checks whether carrier transaction contains either (2/3N + 1) or
 // (N/2 + 1) valid multi-signature of the NeoFS Alphabet.
+// Container contract's `SubmitObjectPut` is an exception, Alphabet signature
+// is not required, contract's `VerifyPlacementSignatures` is called instead.
 func Verify() bool {
-	return common.ContainsAlphabetWitness()
+	cnrH := storage.Get(storage.GetReadOnlyContext(), containerContractKey).(interop.Hash160)
```

Comment:
Can't we use NNS to resolve container contract hash?

Reply:
in fact, it would simplify many-many things. i need an approve that it is not a bad idea compared to `storage.Get`

Reply:
@roman-khimov

Reply:
It depends on the overhead, if we can ensure NNS doesn't add much (it's not obvious right now, the code does many more things to resolve) then OK. Just remember that this code is executed for every single transaction and the result is always the same.

This code however is executed concurrently for every transaction...